feat(public-accessibility-logic): Add AKS and AWS Lambda (#22419)
DataDog/documentation: content/en/security/cloud_security_management/guide/public-accessibility-logic.md

See [Amazon SQS security best practices][33] for more information about public SQS queues.

> ### Amazon Lambda Function

Reviewer: ```suggestion
### AWS Lambda function
```

Reviewer: Thanks for following up. Could you clarify what nomenclature you're referring to?

For additional context about why I've made this request:
 - The [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#headers) requests that level-3 headers use sentence case
 - `Lambda function` is used in AWS documentation (see [examples on this page](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html))
 - `Lambda function` is what's used in other pages of the documentation site (see [examples on this page](https://docs.datadoghq.com/serverless/aws_lambda/installation/python/?tab=datadogcli))

> A [lambda function][58] (`aws_lambda_function`) is considered publicly accessible if:

Reviewer: ```suggestion
A [Lambda function][58] (`aws_lambda_function`) is considered publicly accessible if:
```

Author: Following the nomenclature, I should capitalize both letters.

Reviewer: Thanks for following up. Could you clarify what nomenclature you're referring to?

For additional context about why I've made this request:
 - The [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#headers) requests that level-3 headers use sentence case
 - `Lambda function` is used in AWS documentation (see [examples on this page](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html))
 - `Lambda function` is what's used in other pages of the documentation site (see [examples on this page](https://docs.datadoghq.com/serverless/aws_lambda/installation/python/?tab=datadogcli))

Reviewer: Thanks for following up, I think I accidentally deleted my previous comment. Could you clarify what nomenclature you're referring to?

For additional context about why I'm making this request:
  - The [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#headers) requests sentence case for level-3 headers
  - AWS documentation mostly uses `Lambda function` (see [this page](https://docs.aws.amazon.com/lambda/latest/dg/getting-started.html) for examples)
  - Our documentation mostly uses `Lambda function` (see [this page](https://docs.datadoghq.com/serverless/aws_lambda/logs/?tab=serverlessframework) for examples)

Author: Thanks ! Following your PR, I made the changes that you suggested

> A [Lambda Function][58] (`aws_lambda_function`) is considered publicly accessible if:

Reviewer: ```suggestion
A [Lambda function][58] (`aws_lambda_function`) is considered publicly accessible if:
```

| **Criteria** | **Explanation** |
|--------------|-----------------|

> |The function has a policy that allows any principal (principal_policy or principal_aws set to `"*"`) to perform actions unconditionally.|This setting makes the function accessible to everyone in the world or to any authenticated AWS user.|

Reviewer: ```suggestion
|The function has a policy that allows any principal (`principal_policy` or `principal_aws`) set to `"*"`. |This setting makes the function accessible to everyone in the world or to any authenticated AWS user.|
```

> See [Best practices for working with AWS Lambda Function][59] for more information about public Lambda functions.

Reviewer: ```suggestion
See [Best practices for working with AWS Lambda function][59] for more information about public Lambda functions.
```

Reviewer: Apologies for the back and forth here, but just requesting this minor update for the reasons outlined [here](https://github.com/DataDog/documentation/pull/22419#discussion_r1558047907).

Additionally, if you open the link, you'll find that `function` is not capitalized at the destination, either.

> See [AWS Lambda documentation][59] for more information about public Lambda functions.

Reviewer: ```suggestion
See [Best practices for working with AWS Lambda functions][59] for more information about public Lambda functions.
```

Author: Great changes.
I capitalized function though

To learn more about disallowing blob public access on Azure Storage accounts, see [Choose to allow or disallow blob public access on Azure Storage accounts][46].

> ### Azure AKS Cluster

Reviewer: ```suggestion
### Azure Kubernetes Service (AKS) cluster
```

An [AKS cluster][60] (`azure_aks_cluster`) is considered publicly accessible if:

| **Criteria** | **Explanation** |
|--------------|-----------------|
|`enable_private_cluster` is set to `false` in the cluster's configuration.|This setting makes the cluster publicly accessible when combined with an open public CIDR. |

> |The cluster's `authorized_ip_ranges` contains an open CIDR block (`"0.0.0.0/0"`) or is unset.|You can limit the CIDR blocks that can access the public endpoint of the AKS cluster. An open CIDR block means anyone on the internet can access the endpoint.|

Reviewer: ```suggestion
|The cluster's `authorized_ip_ranges` contains an open CIDR block (`"0.0.0.0/0"`) or is unset.|An open CIDR block means anyone on the internet can access the endpoint.|
```

> See [Azure Kubernetes Service (AKS) clusters][61] for more information on public AKS clusters.

Reviewer: ```suggestion
See [AKS best practices][61] for more information on public AKS clusters.
```
